src/persist-client/src/cache.rs

```diff
@@ -120,10 +133,175 @@ impl PersistClientCache {
     }
 }
 
+trait DynState: Debug + Send + Sync {
```

Comment:
is there a tl;dr on why we need `DynState`? this is far past my ability to reason about rust-isms on paper, but I'm a little surprised about this vs making `StateCache` generic over KVTD

Reply:
I guess we don't really do this in practice (yet?) but otherwise, we'd have to actually carry the K, V, T, D, types all the way up to PersistClient, which really doesn't seem right to me